backend/data/src/main/kotlin/io/tolgee/batch/BatchJobProjectLockingManager.kt

```diff
@@ -71,65 +77,84 @@ class BatchJobProjectLockingManager(
 
   private fun tryLockWithRedisson(batchJobDto: BatchJobDto): Boolean {
     val projectId = batchJobDto.projectId ?: return true
-    val computed =
-      getRedissonProjectLocks().compute(projectId) { _, value ->
-        computeFnBody(batchJobDto, value)
-      }
-    return computed == batchJobDto.id
+    val computedJobIds =
+      getRedissonProjectLocks().compute(projectId) { _, lockedJobIds ->
+        val newLockedJobIds = computeFnBody(batchJobDto, lockedJobIds ?: emptySet())
+        logger.debug(
+          "While trying to lock on redis {} for project {} new lock value is {}",
+          batchJobDto.id,
+          batchJobDto.projectId,
+          newLockedJobIds
+        )
+        newLockedJobIds
+      } ?: emptySet()
+    return computedJobIds.contains(batchJobDto.id)
   }
 
-  fun getLockedForProject(projectId: Long): Long? {
+  fun getLockedForProject(projectId: Long): Set<Long> {
     if (usingRedisProvider.areWeUsingRedis) {
-      return getRedissonProjectLocks()[projectId]
+      return getRedissonProjectLocks()[projectId] ?: emptySet()
     }
-    return localProjectLocks[projectId]
+    return localProjectLocks[projectId] ?: emptySet()
   }
 
-  private fun tryLockLocal(toLock: BatchJobDto): Boolean {
-    val projectId = toLock.projectId ?: return true
-    val computed =
-      localProjectLocks.compute(projectId) { _, value ->
-        val newLocked = computeFnBody(toLock, value)
-        logger.debug("While trying to lock ${toLock.id} for project ${toLock.projectId} new lock value is $newLocked")
-        newLocked
-      }
-    return computed == toLock.id
+  private fun tryLockLocal(batchJobDto: BatchJobDto): Boolean {
+    val projectId = batchJobDto.projectId ?: return true
+    val computedJobIds =
+      localProjectLocks.compute(projectId) { _, lockedJobIds ->
+        val newLockedJobIds = computeFnBody(batchJobDto, lockedJobIds ?: emptySet())
+        logger.debug(
+          "While trying to lock locally {} for project {} new lock value is {}",
+          batchJobDto.id,
+          batchJobDto.projectId,
+          newLockedJobIds
+        )
+        newLockedJobIds
+      } ?: emptySet()
+    return computedJobIds.contains(batchJobDto.id)
   }
 
   private fun computeFnBody(
```

Comment:
maybe we can rename the function to smth more meaningful? Or am i missing the idea behind it?